File: agent-world-lab/backend/src/AgentWorldLab.Infrastructure/Services/ToolCatalogService.cs
using AgentWorldLab.Application.Abstractions;
using AgentWorldLab.Domain.Models;

namespace AgentWorldLab.Infrastructure.Services;

public sealed class ToolCatalogService : IToolCatalogService
{
    private static readonly IReadOnlyCollection<ToolDefinition> Tools =
    [
        new ToolDefinition(
            "time.now",
            "Returns current UTC timestamp for diagnostics and demos.",
            RiskLevel.Low,
            ["Admin", "Approver", "Developer", "Analyst", "Viewer"]),
        new ToolDefinition(
            "aws.cost.snapshot",
            "Returns a lab cost snapshot from AWS adapter path.",
            RiskLevel.Medium,
            ["Admin", "Approver", "Developer", "Analyst"]),
        new ToolDefinition(
            "azure.resource.scan",
            "Scans tagged Azure resources in read-only mode.",
            RiskLevel.High,
            ["Admin", "Approver", "Developer"])
    ];

    public IReadOnlyCollection<ToolDefinition> ListTools() => Tools;

    public ToolDefinition? FindByName(string toolName)
    {
        return Tools.FirstOrDefault(x => string.Equals(x.Name, toolName, StringComparison.OrdinalIgnoreCase));
    }
}

File: agent-world-lab/backend/python/app/infrastructure/services/tool_catalog_service.py
from __future__ import annotations

from app.application.interfaces import IToolCatalogService
from app.domain.models import RiskLevel, ToolDefinition


class ToolCatalogService(IToolCatalogService):
    def __init__(self) -> None:
        self._tools = [
            ToolDefinition("time.now", "Returns current UTC time", RiskLevel.LOW, ["Admin", "Approver", "Developer", "Analyst", "Viewer"]),
            ToolDefinition("aws.cost.snapshot", "Gets mock AWS cost snapshot", RiskLevel.MEDIUM, ["Admin", "Approver", "Developer", "Analyst"]),
            ToolDefinition("azure.resource.scan", "Performs mock Azure resource scan", RiskLevel.HIGH, ["Admin", "Approver", "Developer"]),
        ]

    def list_tools(self) -> list[ToolDefinition]:
        return list(self._tools)

    def find_by_name(self, name: str) -> ToolDefinition | None:
        for tool in self._tools:
            if tool.name.lower() == name.lower():
                return tool
        return None
